lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -610,6 +615,27 @@ export class NormalComponent<
     }
   }
 
+  _doInitialSchematicComponentRenderWithCustomSymbol(
+    symbolElement: ReactElement,
+  ) {
+    if (this.root?.schematicDisabled) return
+    const { db } = this.root!
+
+    const center = this._getGlobalSchematicPositionBeforeLayout()
+
+    const schematic_component = db.schematic_component.insert({
+      center,
+      // width/height are computed in the SchematicComponentSizeCalculation phase
+      size: { width: 0, height: 0 },
+      source_component_id: this.source_component_id!,
+      symbol_name: "custom_symbol",
+      symbol_display_value: this._getSchematicSymbolDisplayValue(),
+    })
+    this.schematic_component_id = schematic_component.schematic_component_id
+
+    this.add(symbolElement)
```

Comment:
i think this is supposed to be added earlier although i could be wrong, adding it so late means it can't participate in port initialization